src/vellum/workflows/nodes/core/map_node/tests/test_node.py

```diff
@@ -172,3 +174,81 @@ class FruitMapNode(MapNode):
         ["apple carrot", "apple potato"],
         ["banana carrot", "banana potato"],
     ]
+
+
+def test_map_node_parallel_execution_with_workflow():
+    # TODO: Find a better way to test this such that it represents what a user would see.
+    # https://linear.app/vellum/issue/APO-482/find-a-better-way-to-test-concurrency-with-map-nodes
+    thread_ids = {}
+
+    # GIVEN a series of nodes that simulate work
+    class BaseNode1(BaseNode):
+        item = MapNode.SubworkflowInputs.item
+
+        class Outputs(BaseOutputs):
+            output: str
+            thread_id: int
+
+        def run(self) -> Outputs:
+            current_thread_id = threading.get_ident()
+            thread_ids[self.item] = current_thread_id
+
+            # Simulate work
+            time.sleep(0.5)
```

Comment:
Is the bug reproducible (ie this test fails) if we do `0.01` instead of `0.5`? would love to keep each test on the order of ms